### PR TITLE
[RayService] Update docs to use multi-app

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Make some small changes to the RayService docs so that it matches with the new multi-application sample rayservice yaml file.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/guidance/rayservice.md [hunk 1]
```diff
@@ -12,12 +12,12 @@
...
 
 * **Ray Cluster**: Manages resources in a Kubernetes cluster.
-* **Ray Serve Deployment Graph**: Manages users' deployment graphs.
+* **Ray Serve Applications**: Manages users' applications.
 
 ### What does the RayService provide?
 
-* **Kubernetes-native support for Ray clusters and Ray Serve deployment graphs.** After using a Kubernetes config to define a Ray cluster and its Ray Serve deployment graphs, you can use `kubectl` to create the cluster and its graphs.
-* **In-place update for Ray Serve deployment graph.** Users can update the Ray Serve deployment graph config in the RayService CR config and use `kubectl apply` to update the deployment graph.
+* **Kubernetes-native support for Ray clusters and Ray Serve applications.** After using a Kubernetes config to define a Ray cluster and its Ray Serve applications, you can use `kubectl` to create the cluster and its applications.
```
Comment (kevin85421) [inline docs/guidance/rayservice.md:19]:
```suggestion
* **Kubernetes-native support for Ray clusters and Ray Serve applications:** After using a Kubernetes config to define a Ray cluster and its Ray Serve applications, you can use `kubectl` to create the cluster and its applications.
```

File: docs/guidance/rayservice.md [hunk 2]
```diff
@@ -12,12 +12,12 @@
...
+* **In-place update for Ray Serve applications.** Users can update the Ray Serve config in the RayService CR config and use `kubectl apply` to update the applications.
```
Comment (kevin85421) [inline docs/guidance/rayservice.md:20]:
```suggestion
* **In-place update for Ray Serve applications:** Users can update the Ray Serve config in the RayService CR config and use `kubectl apply` to update the applications.
```

File: docs/guidance/rayservice.md [hunk 3]
```diff
@@ -12,12 +12,12 @@
...
 * **Zero downtime upgrade for Ray clusters.** Users can update the Ray cluster config in the RayService CR config and use `kubectl apply` to update the cluster. RayService will temporarily create a pending cluster and wait for it to be ready, then switch traffic to the new cluster and terminate the old one.
```
Comment (kevin85421) [inline docs/guidance/rayservice.md:21]:
```suggestion
* **Zero downtime upgrade for Ray clusters:** Users can update the Ray cluster config in the RayService CR config and use `kubectl apply` to update the cluster. RayService will temporarily create a pending cluster and wait for it to be ready, then switch traffic to the new cluster and terminate the old one.
```

File: docs/guidance/rayservice.md [hunk 4]
```diff
@@ -12,12 +12,12 @@
...
 * **Services HA.** RayService will monitor the Ray cluster and Serve deployments' health statuses. If RayService detects an unhealthy status for a period of time, RayService will try to create a new Ray cluster and switch traffic to the new cluster when it is ready.
```
Comment (kevin85421) [inline docs/guidance/rayservice.md:22]:
```suggestion
* **Services HA:** RayService will monitor the Ray cluster and Serve deployments' health statuses. If RayService detects an unhealthy status for a period of time, RayService will try to create a new Ray cluster and switch traffic to the new cluster when it is ready.
```

**Review by kevin85421 — COMMENTED** [review]
1. This doc uses KubeRay v0.5.0 which does not support multi-app.
2. Add some sections for "in-place update" and "zero downtime upgrade". (Update: the doc has in-place update example.)
3. Explain the configurations in `ray_v1alpha1_rayservice.yaml`.

File: docs/guidance/rayservice.md [hunk 5]
```diff
@@ -153,12 +157,12 @@ serveDeploymentStatuses:
 
 After it finishes deployment, let's send a request again. In the curl pod from earlier, run:
 ```shell
-[ root@curl:/ ]$ curl -X POST -H 'Content-Type: application/json' rayservice-sample-serve-svc.default.svc.cluster.local:8000 -d '["MANGO", 2]'
+[ root@curl:/ ]$ curl -X POST -H 'Content-Type: application/json' rayservice-sample-serve-svc.default.svc.cluster.local:8000/fruit/ -d '["MANGO", 2]'
```
Comment (kevin85421) [inline docs/guidance/rayservice.md:160]: We need to tell users why do we need to add `/fruit/` here.

File: docs/guidance/rayservice.md [hunk 6]
```diff
@@ -131,15 +135,15 @@ $ kubectl port-forward service/rayservice-sample-head-svc 8265
 ```
 Access the dashboard using a web browser at `localhost:8265`.
 
-### Update Ray Serve Deployment Graph
+### Update Ray Serve config
```
Comment (kevin85421) [inline docs/guidance/rayservice.md:138]: Users may not know this is equivalent to "In-place update for Ray Serve applications".

**Comment by zcin** [comment]
> 2\. Add some sections for "in-place update" and "zero downtime upgrade". (Update: the doc has in-place update example.)

Can we add this separately? We can track this item in the Serve-Kuberay tasks, my main goal here is to make sure this doc is consistent with the new `ray_v1alpha1_rayservice.yaml`

**Comment by kevin85421** [comment]
> Can we add this separately? We can track this item in the Serve-Kuberay tasks, my main goal here is to make sure this doc is consistent with the new ray_v1alpha1_rayservice.yaml

Sure. Would you mind opening an issue to track the progress? Thanks!

File: docs/guidance/rayservice.md (hunk 5 above)
Comment (zcin) [reply docs/guidance/rayservice.md:160]: I've added explanations!

**Comment by zcin** [comment]
> > Can we add this separately? We can track this item in the Serve-Kuberay tasks, my main goal here is to make sure this doc is consistent with the new ray_v1alpha1_rayservice.yaml
> 
> Sure. Would you mind opening an issue to track the progress? Thanks!

@kevin85421 Created https://github.com/ray-project/kuberay/issues/1184!

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**